src/main/java/org/stratumproject/fabric/tna/behaviour/upf/FabricUpfProgrammable.java

```diff
@@ -547,6 +550,26 @@ private Collection<UpfEntity> getUpfTerminationsDownlink() throws UpfProgrammabl
         return upfTerminations;
     }
 
+    private void applyUpfCounter(UpfCounter upfCounter) throws UpfProgrammableException {
```

Comment:
The UpfProgrammable API allows to do a more generic apply, rather than a simple reset. The same for P4RT, you can do a modify counter and set the counter to 0, there is no explicit way to do a reset. I understand that the only use case we currently have is a reset, but I would keep the API generic and leave the apply, do you have any reason for avoiding this?
Also, the parameter is required, we don't want to reset the entire meter bank, but just a single meter cell.

Reply:
+1 having a more generic API